src/inmanta/deploy/scheduler.py

```diff
@@ -1050,14 +1236,38 @@ async def _get_last_non_deploying_state_for_dependencies(
     def get_types_for_agent(self, agent: str) -> Collection[ResourceType]:
         return list(self._state.types_per_agent[agent])
 
-    async def send_in_progress(self, action_id: UUID, resource_id: ResourceVersionIdStr) -> None:
+    async def send_in_progress(self, action_id: UUID, resource_id: Id) -> None:
         await self._state_update_delegate.send_in_progress(action_id, resource_id)
 
-    async def send_deploy_done(self, result: DeployResult) -> None:
-        return await self._state_update_delegate.send_deploy_done(result)
+    async def send_deploy_done(self, attribute_hash: str, result: DeployResult, deployment_result: DeploymentResult) -> None:
+        await self._state_update_delegate.send_deploy_done(attribute_hash, result, deployment_result)
 
     async def dryrun_update(self, env: uuid.UUID, dryrun_result: executor.DryrunResult) -> None:
         await self._state_update_delegate.dryrun_update(env, dryrun_result)
 
     async def set_parameters(self, fact_result: FactResult) -> None:
         await self._state_update_delegate.set_parameters(fact_result)
+
+    async def update_resource_intent(
```

Comment:
Do we still need to implement the `StateUpdateManager` at all now that we lifted the authority of persisting things from the tasks to the scheduler? I'd think we could simply use composition for the persistence manager. I'd be glad to get rid of all these proxy methods and the extra layer of inheritance.